glitch/analysis/rules.py

```diff
@@ -185,5 +212,5 @@ def check_comment(self, c: Comment, file: str) -> list[Error]:
 
 class SmellChecker(ABC):
     @abstractmethod
-    def check(self, element, file: str) -> list[Error]:
+    def check(self, element, file: str, code = None, elem_name: str = "", elem_value: str = "", au_type = None, parent_name = "") -> list[Error]:
```

Comment:
Adding more attributes to this function will affect maintainability in the future. I don't have the full picture of the code but I can think of two other options:

1. The analyses that require information from "upper" structures of the intermediate representation are done on those structures. For instance: if an analysis requires the atomic unit type, the analysis is done on the atomic unit first and then its attributes are analyzed. This means that if a `SmellChecker` is used, the element it will receive is an atomic unit. The same happens if a parent attribute needs to be analyzed. 
2. A data class is defined to "carry" auxiliary information. I prefer the other option because I don't see a way to define this class in a "clean" way.

Reply:
From what I remember, I did this change because in most of the analyses, it is checked some combinations of attributes, for example if we have a block of attributes, it is necessary the name of the parent attribute so that we can check was is the type of attribute we are dealing with. The type of the atomic unit is used also to identify the attribute we are dealing, because for different atomic units we can have attributes with the same name but with different "secure values".

Reply:
I think most of them require information from upper structures. But they are different. For example: we have cases that is being checked the absence of necessary attributes (this requires the analysis to be done in the atomic unit), and we have cases that we are checking the value of the attributes (this does not need necessarily to be done in the atomic unit, if we pass through the methods, the atomic unit information, but can be done also in the atomic unit and "costs" more).

Reply:
Hmm ok I see, thanks! I'll try to find a way to make this "cleaner".